requirements.txt

```diff
@@ -2,4 +2,4 @@ RadeonOpenCompute/rocm-cmake@839cf61c355f997283e3baaf245fa9acf2e3baa0 --build
 sqlite3@3.17 -DCMAKE_POSITION_INDEPENDENT_CODE=On
 boost@1.72 -DCMAKE_POSITION_INDEPENDENT_CODE=On --build
 half,https://github.com/pfultz2/half/archive/1.12.0.tar.gz -X header -H sha256:0a08660b68abb176ebc2a0cdf8de46e3182a7f46c66443bb80dbfaaec98cf969 --build
-ROCmSoftwarePlatform/llvm-project-mlir@release/rocm-5.2 -DBUILD_FAT_LIBMLIRMIOPEN=1
+ROCmSoftwarePlatform/llvm-project-mlir@59905bc4e56b4d54dd416f00ecf08d1923bfb4e1 -DBUILD_FAT_LIBMLIRMIOPEN=1
```

Comment:
Is it so that this breaks backward compatibility with MLIR from ROCm 5.2?

IIRC MLIR backend is not an experimental feature anymore. Maintenance of backward compatibility with a reasonably selected range of previous versions is a standard requirement for MIOpen.

We can use MLIR version information from at CMake level (and forward it into C++ sources as macros, like we already do for COMGR, for example) to maintain backward compatibility at full, including tests.

Reply:
> Can you please remind me, is MLIR Backend an experimental feature yet?

Now with MIOpen release MLIR by default, it shouldn't be an experimental feature.

> If not, then we have a problem.

I think you are right, It indeed never occurred to me to maintain backward compatibility before.

> We can use MLIR version information from at CMake level

Thanks for the pointer. Let's move the discussion into the ticket you opened. I'll likely need a few more pointers and do whatever necessary.